vault_program/src/create_token_metadata.rs

```diff
@@ -1,11 +1,71 @@
-use solana_program::{account_info::AccountInfo, entrypoint::ProgramResult, pubkey::Pubkey};
+use jito_account_traits::AccountDeserialize;
+use jito_jsm_core::loader::{load_signer, load_system_account, load_system_program};
+use jito_vault_core::{loader::load_vault, vault::Vault};
+use solana_program::{
+    account_info::AccountInfo, borsh1::get_instance_packed_len, entrypoint::ProgramResult,
+    program_error::ProgramError, pubkey::Pubkey, rent::Rent, sysvar::Sysvar,
+};
+use spl_pod::optional_keys::OptionalNonZeroPubkey;
+use spl_token_metadata_interface::state::TokenMetadata;
+use spl_type_length_value::state::TlvStateMut;
 
 pub fn process_create_token_metadata(
-    _program_id: &Pubkey,
-    _accounts: &[AccountInfo],
-    _name: String,
-    _symbol: String,
-    _uri: String,
+    program_id: &Pubkey,
+    accounts: &[AccountInfo],
+    name: String,
+    symbol: String,
+    uri: String,
 ) -> ProgramResult {
+    let [metadata_info, vault_info, vault_admin, system_program] = accounts else {
+        return Err(ProgramError::NotEnoughAccountKeys);
+    };
+
+    load_system_account(metadata_info, true)?;
+    load_vault(program_id, vault_info, false)?;
+    load_signer(vault_admin, true)?;
+    load_system_program(system_program)?;
+
+    let vault_data = vault_info.data.borrow_mut();
+    let vault = Vault::try_from_slice(&vault_data)?;
```

Comment:
missing a vault_admin signer check here

Reply:
I added a new method `check_admin` inside `Vault` to check admin validity.
Similar to here. 
https://github.com/solana-labs/solana-program-library/blob/48b8381f8ac272e27b9425eac6aade669da915dc/stake-pool/program/src/state.rs#L402-L417